src/Client.php

```diff
@@ -316,23 +316,22 @@ public function getCookieJar()
 
     /**
      * @param string $locator The path to an element to be waited for. Can be a CSS selector or Xpath expression.
+     * @param bool $checkVisibility If true means checking that an element is present on the DOM of a page and visible.
      *
      * @throws NoSuchElementException
      * @throws TimeoutException
      *
      * @return Crawler
      */
-    public function waitFor(string $locator, int $timeoutInSecond = 30, int $intervalInMillisecond = 250)
+    public function waitFor(string $locator, int $timeoutInSecond = 30, int $intervalInMillisecond = 250, bool $checkVisibility = false)
```

Comment:
The previous implementation, of having two methods (`waitFor()` and `waitForVisibility()`) is a better choice than the addition of a boolean parameter.

With the introduction of a boolean parameter, this method now does not one thing but two things with the boolean parameter letting you choose which of the two things to execute. A method should do just one thing.

From the calling perspective, the downside to a boolean parameter is that the purpose is not clear when reading code. This can lead to confusing interpretations, make code harder to reason about and makes it easier to introduce bugs.

For the sake of an example, let's simplify things and assume that `waitFor()` takes only two parameters:

```php

public function waitFor(string $locator, bool $checkVisibility): Crawler
{
    // ...
}
```

From reading code that uses this method, it is not possible to determine, without examining the method definition, the difference between these three calls:

```php

$client->waitFor('.selector');
$client->waitFor('.selector', false);
$client->waitFor('.selector', true);
```

Two separate public methods with clear names remove confusion. The method names alone make it clear that there are two different methods achieving two different things:
```php

$client->waitFor('.selector');
$client->waitForVisibility('.selector');
```